static/js/main.js

```diff
@@ -0,0 +1,253 @@
+                }
+
+                maxPages = Math.ceil(filteredData.length / cardsPerPage);
+
+                if (pageCount <= maxPages) {
+                    const moreCountryData = filteredData.slice(startIndex, endIndex);
+                    displayCards(moreCountryData);
+                } else {
+                    maxPagesReached = true;
+                }
+
+                displayLoader(false);
+            });
+
+    }
+}
+
+const displayCards = (countries) => {
+    const cards = [];
+
+    countries.forEach((country) => {
+
+        let countryCapital = country.capital;
+
+        if (countryCapital === undefined) {
+            countryCapital = 'None';
+        }
+
+        const card = document.createElement('div');
+        card.classList.add('country-card');
+        card.innerHTML = ` 
+        <img class="country-flag" src="${country.flag}">
+        <h2 class="country-name">${country.name}</h2>
+        <ul class="card-ul">
+            <li class="card-li">
+                <span class="span-category">Population:</span>
+                <span class="span-value">${country.population.toLocaleString('en-US')}</span>
+            </li>
+            <li class="card-li">
+                <span class="span-category">Region:</span>
+                <span class="span-value">${country.region}</span>
+            </li>
+            <li class="card-li">
+                <span class="span-category">Capital:</span>
+                <span class="span-value">${countryCapital}</span>
+            </li>
+        </ul>
+        `;
+
+        cards.push(card);
+    });
+
+    cards.forEach((card) => {
+        countryCardContainer.appendChild(card);
+    });
+}
+
+const addRegionsToSelectDropdown = () => {
+    const regions = [];
+    fetchData(['region'])
+        .then((data => {
+            data.forEach((country) => {
+                if (!regions.includes(country.region)) {
+                    regions.push(country.region);
+                }
+            });
+
+            regions.forEach((region) => {
+                const li = document.createElement('li');
+                const input = document.createElement('input');
+                const label = document.createElement('label');
+
+                input.setAttribute('type', 'radio');
+                input.setAttribute('id', region.toLowerCase());
+                input.setAttribute('class', 'dropdown-input');
+                input.setAttribute('name', 'region');
+                label.textContent = region;
+                label.setAttribute('for', region.toLowerCase());
+
+                li.appendChild(input);
+                li.appendChild(label);
+                selectDropDown.appendChild(li);
+            });
+        }));
+}
+
+const setFilter = (event) => {
+    event.preventDefault();
+    pageCount = 1;
+    maxPagesReached = false;
+
+    filterActive = true;
+    searchActive = false;
+
+    let input = event.target;
+    if (input.tagName === 'LI') {
```

Comment:
Waarom dient dit stukje. Line 204 - 208. 

Als dit te maken heeft met je click event, wat soms op de input zit, maar soms op de label. Dan zou ik het misschien anders aanpakken. Je zou een manier kunnen zoeken, in de html en/of css, om ervoor te zorgen dat als je op een dropdown item clicked, dat de event consistent is elke keer.